river/sync.go

```diff
@@ -209,16 +226,31 @@ func (r *River) makeUpdateReqData(req *elastic.BulkRequest, rule *Rule,
 	req.Action = elastic.ActionUpdate
 
 	for i, c := range rule.TableInfo.Columns {
+		mapped := false
 		if reflect.DeepEqual(beforeValues[i], afterValues[i]) {
 			//nothing changed
 			continue
 		}
-		if name, ok := rule.FieldMapping[c.Name]; ok {
-			// has custom field mapping
-			req.Data[name] = r.makeReqColumnData(&c, afterValues[i])
-		} else {
+		for _, m := range rule.FieldMapping {
+			if m.Mysql == c.Name {
+				mapped = true
+				// has custom field mapping
+				v := r.makeReqColumnData(&c, afterValues[i])
+				str, err := v.(string)
+				if err == false {
+					req.Data[c.Name] = v
```

Comment:
here should using continue, otherwise how to handle v is not string but type is list?